template/template_test.go

```diff
@@ -506,6 +506,16 @@ func TestTemplateFuncs(t *testing.T) {
 		title: "Template using reReplaceAll",
 		in:    `{{ reReplaceAll "ab" "AB" "abc" }}`,
 		exp:   "ABc",
+	}, {
```

Comment:
Can you please add unhappy cases? I'm not sure what happens when you pass a non-valid format to `date` or when empty value `time` and an error.

Reply:
Yeah sure! Template panics like other functions. As far as I can tell, there is no discernible difference between returning an error and panicking in the function, both result in a panic:

```go
"reReplaceAll": func(pattern, repl, text string) string {
	re := regexp.MustCompile(pattern) // panics if regex is invalid
	return re.ReplaceAllString(text, repl)
},
```

```go
// tz returns the time in the timezone.
"tz": func(name string, t time.Time) (time.Time, error) {
	loc, err := time.LoadLocation(name)
	if err != nil {
		return time.Time{}, err // panics with err
	}
	return t.In(loc), nil
},
```